### ISSUE TITLE
[🔥🐛] @react-native-firebase/dynamic-links ios issues

### ISSUE THREAD

**Issue by SohelIslamImran** [invertase/react-native-firebase#6884]
## Issue
`dynamicLinks().getInitialLink()`  and `dynamicLinks().onLink()` do not being called or return undefine on IOS. But on android, it is working perfectly.
Related issues: https://github.com/invertase/react-native-firebase/issues/3450, https://github.com/invertase/react-native-firebase/issues/4548, https://github.com/invertase/react-native-firebase/issues/5962 
`Linking.addEventListener` from `"expo-linking"` receives the dynamic short url but does not the deep link. So I tried to parse the  dynamic link/url with `dynamicLinks().resolveLink(link)`, On android it is successfully parsing to deep link, but on ios the same link is throwing this error when parsing  with `dynamicLinks().resolveLink(link)` :  
```sh
[dynamiclinks/resolve-link-error] the operation couldn't be completed. Universal link url cloud not be parsed by Dynamic links
 ```
I'm using Expo dev client and eas build. The [`@react-native-firebase/dynamic-links`](https://rnfirebase.io/dynamic-links/usage#ios-setup) [config plugin](https://rnfirebase.io/#managed-workflow) is not working as mentioned here https://rnfirebase.io/dynamic-links/usage#ios-setup

Also, the [example](https://cdn.discordapp.com/attachments/1070487687527669760/1070487687775141968/image.png) given here https://rnfirebase.io/dynamic-links/usage#create-a-link does not work. It creates [web links](https://cdn.discordapp.com/attachments/1070487687527669760/1070487688005812234/image.png) instead of [deep links for each platform](https://tilleli.com/event/ziVmGM5aNcMQ3mvu9?d=1). It requires providing [android and ios package and bundle info](https://cdn.discordapp.com/attachments/1070487687527669760/1070796838401216512/image.png), but I thought it will get the info from `GoogleService-Info.plist` and `google-services.json` files. The documentation does not provide any info that we need to provide these.

Here are some my configs:
```json
// app.json
{
  "expo": {
    "scheme": "customscheme",
    ...
    "ios": {
      "usesAppleSignIn": true,
      "googleServicesFile": "./services/GoogleService-Info.plist",
      "associatedDomains": ["applinks:custom-domain.com"],
      "infoPlist": {
        "FirebaseDynamicLinksCustomDomains": ["https://custom-domain.com" ],
        "NSUserActivityTypes": ["INSendMessageIntent"],
        "UIBackgroundModes": ["remote-notification"],
        "LSApplicationQueriesSchemes": ["customscheme"]
      },
      "bundleIdentifier": "com.custom.app",
      "buildNumber": "1.0.5"
    },
    "plugins": [
      [
        "expo-build-properties",
        {
          "ios": {
            "useFrameworks": "static"
          }
        }
      ],
      "expo-apple-authentication",
      "./react-native-maps-plugin",
      "@react-native-firebase/app",
      "@react-native-firebase/crashlytics",
      "@react-native-firebase/dynamic-links"
    ],
    ...
  }
}
```

---

### Javascript

<details><summary>Click To Expand</summary>
<p>

#### `package.json`:

```json
    "expo": "~47.0.13",
    "@react-native-firebase/analytics": "^16.7.0",
    "@react-native-firebase/app": "^16.7.0",
    "@react-native-firebase/auth": "^16.7.0",
    "@react-native-firebase/crashlytics": "^16.7.0",
    "@react-native-firebase/dynamic-links": "^16.7.0",
```
</p>
</details>

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
System:
    OS: Windows 10 10.0.22621
    CPU: (12) x64 AMD Ryzen 5 5600G with Radeon Graphics
    Memory: 3.12 GB / 13.90 GB
  Binaries:
    Node: 18.13.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.19.3 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK: Not Found
  IDEs:
    Android Studio: AI-221.6008.13.2211.9477386
    Visual Studio: Not Found
  Languages:
    Java: 17.0.6 - /c/Program Files/Common Files/Oracle/Java/javapath/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.1.0 => 18.1.0
    react-native: 0.70.5 => 0.70.5
    react-native-windows: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
    
    ---------
expo-env-info 1.0.5 environment info:
    System:
      OS: Windows 10 10.0.22621
    Binaries:
      Node: 18.13.0 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
      npm: 8.19.3 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: AI-221.6008.13.2211.9477386
    npmPackages:
      expo: ~47.0.13 => 47.0.13
      react: 18.1.0 => 18.1.0
      react-native: 0.70.5 => 0.70.5
    Expo Workflow: bare
```

- **Platform that you're experiencing the issue on**:
  - [x] iOS
  - [ ] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **`react-native-firebase` version you're using that has this issue:**
  - `16.7.0`
- **`Firebase` module(s) you're using that has the issue:**
  - `@react-native-firebase/dynamic-links`
- **Are you using `TypeScript`?**
  - `NO`
</p>
</details>

**Comment by SohelIslamImran** [comment]
You need to add documentation about expo here.

![image](https://user-images.githubusercontent.com/76745339/216787342-e6c695f6-5906-429d-956f-1150b7409a33.png)

**Comment by mikehardy** [comment]
Expo support is provided by the community, stated differently, please propose a PR to the documentation - github web UI makes it easy to do so, but as none of the maintainers use Expo we don't know what to add. Someone that uses Expo will need to do it

**Comment by dochonglo** [comment]
Hey @SohelIslamImran - I'm having the same issue on the Expo Managed Workflow. I followed the exact steps you did for installation and integration. Unfortunately, deep links that are created don't route the user back to my app. However, I have the same app running on the Bare Workflow (and I followed the integration guides on rnfirebase.com) and the deep links all work. What might be the issue with the Firebase deep links not working on the Managed Workflow?

**Comment by SohelIslamImran** [comment]
@dochonglo Make sure you have properly provided these two config in your [`app.json`](https://docs.expo.dev/versions/latest/config/app). 

  >  ```json
  >   "associatedDomains": ["applinks:custom-domain.com"],
  >    "infoPlist": {
  >      "FirebaseDynamicLinksCustomDomains": ["https://custom-domain.com" ],
  > ```
  
For me, I did not provide [`FirebaseDynamicLinksCustomDomains`](https://react-native-firebase-m4mspau6w-invertase.vercel.app/dynamic-links/usage#dynamic-links-with-custom-domains) key first time which was cause the issue.

**Comment by kbwhodat** [comment]
I've been literally working on this all day and finally got it using expo. Don't use `expo-linking`. I did the same thing. iOS is unable figure out what the "deep link" is compared to Android so whatever url you press (Ex. http://example.com/foo) will be return as-is to you ios app. I switched to using the dynamicLinks library only. This will handle if the app is running in the background:
```
     const handleDynamicLink = async (link) => {
	  const { url } = link
	  console.log("DYNAMIC", url)
	  const test = await dynamicLinks().resolveLink(url)
	  console.log("RESOLVE", test)
	};

      React.useEffect(() => {
        const unsubscribe = dynamicLinks().onLink(handleDynamicLink);
        // When the component is unmounted, remove the listener
        return () => unsubscribe();
      }, []);

```

The error you got: `[dynamiclinks/resolve-link-error] the operation couldn't be completed. Universal link url cloud not be parsed by Dynamic links` I saw it when I thought I had to resolve the initial link (`dynamicLInks().initialLink()`). Looks like this returns the deep link url for you and there's no need to resolve that link.
